feat(data): refactor KALI data to use legixplore API

adipasquale opened the pull request:
implements #533

In this PR I did a lot of cleanups of deprecated code linked to the conventions. Also, we're loosing all the data that maps APE codes to IDCCs. We'll see if we want to bring it back, but probably we're going to rely more on SIRENs. 

I'm preserving a single `kali.json` data file, that only contains what comes from the newly deployed https://api.kali.num.social.gouv.fr/. It's almost a pure dump of the conteneurs routes on this API, just adding an URL pointing to legifrance.

Visible Changes: 
- The new KALI slugs look like `1930-nouvelle-convention-collective-nationale-des-metiers-de-la-transformation-des-gr` ie the idcc number and the slug of the title truncated to 80 characters. 
- The links to Legifrance are fixed, they used to point to the index page
- Small improvements on the IDCC suggest box results

⚠️ @lionelB Locally I could not make search work with the conventions. But it doesn't work on the master branch locally either so I'm guessing it's a local only problem. We should really double check it before merging.

lionelB commented on packages/code-du-travail-data/dataset/kali/index.js:
```diff
@@ -1,50 +1,19 @@
-    console.error(error);
-  }
-}
+const axios = require('axios');
```
why not keep `node-fetch` ?

adipasquale replied:
I'm just more used to axios, but I can switch back to node-fetch.
also, node-fetch has 3k stars on GitHub vs 55k for axios 🚀

adipasquale replied:
do you want me to revert to node-fetch @lionelB ? I don't have a strong opinion on this.

lionelB commented on packages/code-du-travail-data/dataset/kali/package.json:
```diff
@@ -1,15 +1,13 @@
 {
   "name": "@cdt/data...kali",
-  "version": "1.8.0",
+  "version": "2.0.0",
```
les version des packages du mono-repo sont gérés par la CI 
en gros on utilise un standard de nommage des commit (angular conventionnal commit) pour gérer automatiquement les changements de versions

adipasquale replied:
ah ok je vais revert ca, merci @lionelB . 
je ne comprends pas encore tout a cette histoire de conventional commits

lionelB replied:
pour faire simple, "conventional commit" part du principe que tu ajoutes des métadonnées à tes messages de commit en fonction des modif apportées.
- un bugfix incrémentera le patch `fix(data): update kali data`
- une nouvelle feature incrémentera la mineur `feat(data): use data from kali api`
-  une feat qui ajoute un `breaking change` (on met breaking change dans le message de commit)  va incrémenter la majeur 
```
feat(data): use new data from api

BREAKING CHANGE:  json payload updated
```

https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary

adipasquale replied:
👌 merci @lionelB

adipasquale replied:
du coup je vais commit --amend et push force sur cette branche pour modifier mon commit (rajouter le BREAKING CHANGE et enlever la modif manuelle de la version), ok @lionelB ?

lionelB replied:
Ça m’a paraît bien !

lionelB replied:
une fois tes modif faite, ca t'irais si je rajoute un commit ou 2 pour modifier l'api / front ?

adipasquale replied:
(je repete juste a l'ecrit ce qu'on s'est dit a l'oral:) l'idee est plutot que je fasse moi meme les modifs sur cette PR

adipasquale committed:
feat(data): use new KALI data from legixplore API and remove deprecated data

BREAKING CHANGE: json payload updated

lionelB commented on packages/code-du-travail-data/search/indexing/cdtn_documents.py:
```diff
@@ -48,41 +48,20 @@ def make_slug(text, seed):
                 'slug': slugify(val['titre'], to_lower=True),
                 'title': val['titre'],
-                'all_text':  f"{('IDCC ' + val['idcc']) if 'idcc' in val else 'TI'} {val['titre']}",
+                'all_text':  f"{val["nature"]} {val['num']} {val['titre']}",
```
que contient nature ? 
`all_text` n'etant plus utilisé pour l'indexation, une pr est en train de netoyer son usage. 
par contre si il y a du text sur lesquel on souhaiterai faire des recherches, autant utiliser une champ `text` (mais pas comme all_text avec un dump de tout les champ texte)

adipasquale replied:
merci pour tes retours @lionelB. `nature` contient 'IDCC' ou 'TI'. on veut faire des recherches sur le titre oui. 
si je comprends bien je change comme ca : 
```
{ 
  text: titre,
  nature: nature, 
  num: num
}
```
c'est ca ?

adipasquale replied:
ping @lionelB ^

lionelB replied:
excuse moi j'avais préparer une réponse hier et avant de la mettre je voulais avoir savoir la diff entre id et num

lionelB replied:
```
            CDTN_DOCUMENTS.append({
                'source': 'kali',
                'id': val['id'],
                'slug': slugify(val['titre'], to_lower=True),
                'title': val['titre'],
                'url': val['url'],
                'idcc': val['num]),
            })
```

adipasquale replied:
🆗 merci !

lionelB commented on packages/code-du-travail-data/dataset/kali/package.json:
```diff
@@ -5,11 +5,9 @@
   "license": "MIT",
   "scripts": {
-    "start": "node index.js > idcc-kali-ape.json"
+    "start": "node index.js > ../kali.json"
```
j'avais commencé à replacer les fichier de données dans leur propre répertoire pour éviter d'avoir trop de fichier à la racine de dataset.

adipasquale replied:
ah ok j'ai cru que c'etait l'inverse en regardant les autres ! je vais changer

lionelB commented on packages/code-du-travail-data/dataset/kali/utils.js:
```diff
@@ -1,36 +0,0 @@
-async function batchPromise(list, size, callback) {
```
cette fonction est utilisée dans le projet `@cdt/data...fiches_ministere_travail`
que j'avais prévu de le remplacer par la lib [p-limit](https://github.com/sindresorhus/p-limit)
Je propose que tu le laisses pour  l'instant et je ferai un PR pour le supprimer et refactoriser l'autre projet

adipasquale replied:
woups pardon je suis allé trop vite, bien vu merci

lionelB committed:
fix(api): update search query for kali

we duplicate title in text in order to make multimatch query work on
signle term search since search is multimatch

lionelB committed:
fix(api): update idcc search to use all title fields

lionelB commented:
@adipasquale la recherche marchait pas non plus chez moi. On utilise un multimatch sur le titre et le text or dans le ccn, on a que le titre. En renseignant un champ text avec le titre de la convention cela fonctionne mieux.
À confirmer auprès du consultant elastic

J'ai rajouté les 2 commit, il ne reste plus qu'à  rebaser la PR ? (j'ai testé un rebase en local, pas de conflits)

adipasquale commented:
bien vu pour la recherche @lionelB merci je corrige ca vendredi premiere chose et je rebase la PR. merci !

adipasquale committed:
fixes for KALI: ES indexing improvements, moved kali.json file, restored utils

adipasquale committed:
updated idcc search test snapshot

adipasquale committed:
removed highlighting from ES query for IDCC
